src/main/java/duke/commands/CommandResult.java

```diff
@@ -0,0 +1,52 @@
+package src.main.java.duke.commands;
+
+import java.util.List;
+import java.util.Optional;
```

Comment:
Really like that you consider using Java's `Optional`s to pre-empt empty tasks! Most people – including me – would stray away from this and bulk up on error handling instead. Kudos!